### ISSUE TITLE
loop variable access in loop local function gives null

### ISSUE THREAD

**Issue by JanCervak** [scriban/scriban#385]
In my scenario I would like to filter an array with a loop variable. It seems that functions can be defined inside a loop, but it seems that the loop variable is not available to be used in the function (the value is just null as it were not defined). Is the function limited to read just function arguments? In the doc I've found just a note that 'simple function' can set a global variable.

The following sample gets arrays `Parameter[] CommonParameters` and `Method[] AllMethods` from object model and uses an imported `static bool HasMethodParameter(Method, Parameter)`.

```
for p in CommonParameters
    func HasMethodCurrentParameter
        ret HasMethodParameter $0 p
    end
    containingMethods = AllMethods | array.filter @HasMethodCurrentParameter
    ...
```

The called .NET HasMethodParameter() gets null for the _p_ variable. Is this intended behaviour?

**Comment by xoofx** [comment]
Loop variables are special as they are local to the loop and they don't persist outside of it.
If you need to access a loop variable you need to assign it to another variable

**Comment by xoofx** [comment]
Actually, reconsidering this, I fixed it in 402039e and it will be available in Scriban 5.x+

I think the behavior was changed at some point in 4.x but it made the scope of the loop variable more confusing.
Now it should be similar to standard variable, global or local (prefixed by `$`) should work as expected. The only difference for loop variable is that they don't survive after the loop, which make them still special: they are temporary global or local for the life of the loop
